### ISSUE TITLE
Opinion on noised clip

### ISSUE THREAD

**Issue by rom1504** [lucidrains/DALLE2-pytorch#104]
![Screenshot_2022-05-20-12-48-56-151_com google android apps docs](https://user-images.githubusercontent.com/2346494/169515129-2aeba478-183f-4b68-b38d-1b9e3dba6e70.jpg)
![Screenshot_2022-05-20-12-56-53-431_com google android apps docs](https://user-images.githubusercontent.com/2346494/169515135-37478905-1141-4eb4-82dd-4bfee8bc64dd.jpg)

Glide and dalle2 use a noised clip 
Openai reports in dalle2 paper that it improves things a bit compared to vanilla clip

What do you think about this? Is that noised clip important ? Could it have more properties beyond improving diffusion ? (IE could it improve robustness for zero shot / retrieval/ clip guiding)

**Comment by nousr** [comment]
> Could it have more properties beyond improving diffusion ? (IE could it improve robustness for zero shot / retrieval/ clip guiding)

In my experience with adversarial training, a little bit of training on (even slight) perturbations in an image can greatly improve your ability to perform on adversarial samples. You may see a slight decrease in performance in the higher-percentiles of accuracy, but overall I would expect an a more robust model. 

It's probably worth noting, however, that slight perturbations aren't as aggressive as a x_t noised images. It is unclear what exactly they mean by "noise-aware CLIP" ... but If you had some way of informing CLIP what noise-level you are at  (i,e, an additional input token) you may be able to generalize to both (heavier noised) clip-guidance tasks, as well as (lower-noise) tasks like zero-shot and retrieval.

I could imagine such a large contrastive model like CLIP would have incredibly sensitive gradients. As a result the biggest gain, as mentioned in the GLIDE paper, would be to have the proper gradients as you guide the diffusion process. Those early steps are crucial for landing you in the proper image space. Undoubtedly, this is why they continued to use the noised clip from GLIDE in DALLE2 as they probably found it necessary to taking the performance of clip-guided diffusion to the next level.

**Comment by nousr** [comment]
> the current designs of denoising the image embeddings that we see as the diffusion prior

nice catch, I didn't even make that connection--I think you might be right!

**Comment by rom1504** [comment]
I think we got this answered and don't need to do any action on this here
